### PR TITLE
Rename Bunch keys

### PR DESCRIPTION
**Pull request by YanisLalou**
#25 
- Rename Bunch keys in the `DomainAwareDataset` def

### CODE REVIEW

File: skada/datasets/_base.py [hunk 1]
```diff
@@ -232,8 +232,8 @@ def pack(
         target = np.concatenate(ys)
         sample_domain = np.concatenate(sample_domains)
         return (data, target, sample_domain) if return_X_y else Bunch(
```
[the remark this answers is not in the document]
Comment (YanisLalou) [reply skada/datasets/_base.py]: To avoid confusion with the X and y in the for loop, I reuse the names Xs and ys instead of data and target respectively

**Comment by kachayev** [comment]
It seems those we mentioned in the docs as well (in the README file).

**Comment by kachayev** [comment]
Plus sample generator functions have documentation (not sure if they implement packing locally but worth checking out).

**Comment by kachayev** [comment]
The documentation for `pack_lodo` method of the base dataset has the same set of keys as well.

**Comment by kachayev** [comment]
Same for Office31 datasets.

**Comment by kachayev** [comment]
It seems the `_load_office31` loader for the Office31 dataset still returns keys with old names.

File: skada/datasets/_office.py [hunk 2]
```diff
@@ -470,4 +470,7 @@ def _load_office31(
             data.append(content.astype(loader_spec.dtype))
     files['data'] = np.vstack(data)
```
Comment (rflamary) [inline skada/datasets/_office.py:471]: just set X and y here no need to add stuff and pop it

Comment (YanisLalou) [reply skada/datasets/_office.py:471]: We can't. The `files` object comes from the `load_files` function, which is a sklearn function. Thus the keys 'data' and 'target' exists from the beginning.
Thus at some point we would have to remove these keys to not have duplicates with the keys 'X' and 'y'.
Thus the pop methods

**Review by kachayev — APPROVED** [review]

**Pull request merged**